Add BigCommerce Gatsby Netlify CMS example

becomevocal opened the pull request:
**- Summary**

Adds the BigCommerce Gatsby Netlify CMS example that will be demo'd during JAMStack SF 2019.

**- Test plan**

Normal yarn, yarn start method of running this repo.

**- Description for the changelog**

New template md file and associated preview image for this example.

**-Screenshot (Proof)**
<img width="1772" alt="Screen Shot 2019-10-16 at 1 59 24 PM" src="https://user-images.githubusercontent.com/2677921/66950029-3e36b000-f01d-11e9-8621-f8609e144339.png">

safoo commented:
@philhawksworth please take a look

erquhart commented:
Why overwrite the existing e-commerce template?

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

becomevocal commented:
@erquhart wasn't going for that. Used it as a reference while building locally and forgot to split out the file. My bad!

I've reverted that in this PR and split out the new template as intended.
